lib/apiservers/engine/backends/system.go

```diff
@@ -288,6 +288,10 @@ func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.A
 		registryAddress = registryAddress + "/v2/"
 	}
 
+	if !strings.HasPrefix(registryAddress, "https://") && !strings.HasPrefix(registryAddress, "http://") {
+		registryAddress = "https://" + registryAddress
```

Comment:
Can't imagine that we need to support any protocol other than http or https in this case.  However, this is at least the 4th place where tack on a default scheme.  The other 3 that I know of are:
 
* govmomi's soap.ParseURL 

* vic/lib/install/validate/validator.go:ParseURL

* vic/pkg/flags/url_flag.go:Set

For now maybe just add a "TODO: ..." comment, that we could use common method to deal with this.